vllm/v1/sample/rejection_sampler.py

```diff
@@ -245,25 +246,85 @@ def compute_probs(
         return logits
 
     num_tokens = logits.shape[0]
-    batch_size = cu_num_draft_tokens.shape[0]
-    expanded_temperature = torch.empty(
-        (num_tokens, 1),
-        dtype=torch.float32,
-        device=logits.device,
-    )
-    expand_kernel[(batch_size, )](
-        expanded_temperature,
+    temperature = expand_batch_to_tokens(
         sampling_metadata.temperature,
         cu_num_draft_tokens,
-        GREEDY_TEMPERATURE,  # replace_from
-        1,  # replace_to
-        MAX_NUM_TOKENS=MAX_SPEC_LEN,
-        num_warps=1,
+        num_tokens,
+        replace_from=GREEDY_TEMPERATURE,
+        replace_to=1,
     )
-    output_prob = compiled_softmax(logits, expanded_temperature)
+    # TODO(woosuk): Consider using in-place op to reduce memory usage.
+    logits = logits / temperature.unsqueeze(-1)
```

Comment:
inplace: 

logits.div_(temparture.unsqueeze(-1))

Reply:
I'm a bit hesitant because if we do it, we need to take care of the side effect. Basically making sure no other op after rejection sampler uses the (updated) logits.

Reply:
@njhill Good point. After second thought, I actually planned to do it in a followup PR. Let's do it in the next PR!